tests/fixtures/contract-fixtures.ts

```diff
@@ -0,0 +1,28 @@
+//import { contractSources } from "./tests/constants/contractSources";
+//import { getCompiled } from "./tests/util/contracts";
+
+import { contractSources } from "../tests/constants/contractSources";
+import { getCompiled } from "../tests/util/contracts";
+
+// can be async or not
+// export async function mochaGlobalSetup() {
+//   // First compile all contracts
+//   console.log("Making sure all contracts are compiled...");
+//   let keys = Object.keys(contractSources);
+//   for (let i = 0; i < keys.length; i++) {
+//     await getCompiled(keys[i]);
+//   }
+//   console.log("done");
+//   process.exit();
+// }
+// mochaGlobalSetup();
+
+exports.mochaGlobalSetup = async function () {
+  // First compile all contracts
+  console.log("Making sure all contracts are compiled...");
+  let keys = Object.keys(contractSources);
+  for (let i = 0; i < keys.length; i++) {
+    await getCompiled(keys[i]);
```

Comment:
Btw, it would be more efficient instead of:
```
for (let i = 0; i < keys.length; i++) {
    await getCompiled(keys[i]);
}
```
to do:
```
await Promise.all(Object.keys(contractSources).map(getCompiled);
```

It would do concurrent execution in that case, and we know that there cannot be 2 same keys so there won't be conflict/race-condition running all at the same time

Reply:
ok let me try. I was afraid of race conditions indeed